ui/pages/remote-mode/setup/setup-daily-allowance/remote-mode-setup-daily-allowance.component.tsx

```diff
@@ -169,27 +171,39 @@ export default function RemoteModeSetupDailyAllowance() {
       return;
     }
 
-    const newAllowance = {
-      tokenType: selectedAllowanceToken,
+    const selectedAsset = assetsWithBalance.find(
+      (asset) => asset.address === selectedAllowanceAddress,
+    ) as TokenInfo;
+
+    // TODO: handle error
+    if (!selectedAsset) {
+      return;
+    }
+
+    const newAllowance: DailyAllowance = {
       amount: parseFloat(dailyLimit),
-      iconUrl: TOKEN_DETAILS[selectedAllowanceToken].iconUrl,
+      address: selectedAsset.address,
+      image: TOKEN_DETAILS[selectedAsset.symbol].image,
+      name: selectedAsset.name,
+      symbol: selectedAsset.symbol,
+      type: selectedAsset.type,
     };
 
     setDailyAllowance((prevAllowances) => {
       const filteredAllowances = prevAllowances.filter(
-        (allowance) => allowance.tokenType !== selectedAllowanceToken,
+        (allowance) => allowance.address !== selectedAllowanceAddress,
       );
       return [...filteredAllowances, newAllowance];
     });
 
-    setSelectedAllowanceToken(TokenSymbol.ETH);
+    setSelectedAllowanceAddress(TokenSymbol.ETH);
```

Comment:
This looks like it could be slightly problematic given it's storing the enum value here and the address of the asset elsewhere. Could be useful to be more specific with the type to mitigate this (or use union if both are valid)?

Reply:
I see. I changed that to an empty string. The native token asset data for address doesn't have any address at all. This should fix for now. The following PRs will introduce the native token address to be set to `0x0000000000000000000000000000000000000000`